tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/resolution/CascadingValueExtactorResolutionAlgorithmTest.java

```diff
@@ -0,0 +1,215 @@
+import org.hibernate.beanvalidation.tck.tests.valueextraction.resolution.model.IWrapper11;
+import org.hibernate.beanvalidation.tck.tests.valueextraction.resolution.model.IWrapper111;
+import org.hibernate.beanvalidation.tck.tests.valueextraction.resolution.model.IWrapper21;
+import org.hibernate.beanvalidation.tck.tests.valueextraction.resolution.model.IWrapper211;
+import org.hibernate.beanvalidation.tck.tests.valueextraction.resolution.model.IWrapper212;
+import org.hibernate.beanvalidation.tck.tests.valueextraction.resolution.model.IWrapper22;
+import org.hibernate.beanvalidation.tck.tests.valueextraction.resolution.model.IWrapper221;
+import org.hibernate.beanvalidation.tck.tests.valueextraction.resolution.model.Wrapper2;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.audit.annotations.SpecAssertion;
+import org.jboss.test.audit.annotations.SpecVersion;
+import org.testng.annotations.Test;
+
+/**
+ * @author Guillaume Smet
+ */
+@SpecVersion(spec = "beanvalidation", version = "2.0.0")
+public class CascadingValueExtactorResolutionAlgorithmTest extends AbstractTCKTest {
+
+	@Deployment
+	public static WebArchive createTestArchive() {
+		return webArchiveBuilder()
+				.withTestClass( CascadingValueExtactorResolutionAlgorithmTest.class )
+				.withPackage( IWrapper11.class.getPackage() )
+				.build();
+	}
+
+	@Test
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_VALIDATIONROUTINE_VALUEEXTRACTORRESOLUTION_ALGORITHM_CASCADED, id = "a")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_VALIDATIONROUTINE_VALUEEXTRACTORRESOLUTION_ALGORITHM_CASCADED, id = "b")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_VALIDATIONROUTINE_VALUEEXTRACTORRESOLUTION_ALGORITHM_CASCADED, id = "d")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_VALIDATIONROUTINE_VALUEEXTRACTORRESOLUTION_ALGORITHM_CASCADED, id = "e")
+	public void mostSpecificValueExtractorFound() {
+		Validator validator = Validation.byDefaultProvider()
+				.configure()
+				.addValueExtractor( new IWrapper11ValueExtractor0() )
+				.addValueExtractor( new IWrapper111ValueExtractor0() )
+				.addValueExtractor( new IWrapper111ValueExtractor1() )
+				.buildValidatorFactory()
+				.getValidator();
+
+		Set<ConstraintViolation<CascadingEntity1>> violations = validator.validate( new CascadingEntity1( null ) );
+
+		assertThat( violations ).containsOnlyPaths(
+				pathWith()
+						.property( "wrapper" )
+						.property( "property", false, null, null, IWrapper111.class, 0 )
+		);
+	}
+
+	@Test(expectedExceptions = ConstraintDeclarationException.class)
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_VALIDATIONROUTINE_VALUEEXTRACTORRESOLUTION_ALGORITHM_CASCADED, id = "c")
+	public void customGenericTypeWithCascadingButNoValueExtractorThrowsException() {
```

Comment:
Yeah, the issue is not the method but the class used, it should have been `CascadingEntity1`. Fixed.